tests/src/test/java/io/openshift/booster/OpenshiftIT.java

```
@@ -0,0 +1,168 @@
+        changeNameServiceState(NAME_OK);
+        System.out.println("Sleep for " + SLEEP_WINDOW + " ms (circuitBreaker.sleepWindowInMilliseconds)");
+        Thread.sleep(SLEEP_WINDOW);
+        assertGreeting(HELLO_OK);
+        assertCircuitBreaker(CB_CLOSED);
+    }
+
+    private void assertGreeting(String expected) {
+        Response response = RestAssured.when().get(greetingBaseUri + "/api/greeting");
+        System.out.println("/api/greeting " + response.getBody().asString());
+        response.then().statusCode(200).body(containsString(expected));
+    }
+
+    private void assertCircuitBreaker(String expected) {
+        Response response = RestAssured.when().get(greetingBaseUri + "/api/cb-state");
+        System.out.println("/api/cb-state " + response.getBody().asString());
+        response.then().statusCode(200).body(containsString(expected));
+    }
+
+    private void changeNameServiceState(String expected) {
+        Response response = RestAssured.given().header("Content-type", "application/json").body(expected).put(nameBaseUri + "/api/state");
+        response.then().assertThat().statusCode(200).body(equalTo(expected));
+        System.out.println("/api/state " + response.getBody().asString());
+    }
+
+    /**
+     *
+     * @param name
+     * @param templatePath
+     * @return the app route
+     * @throws IOException
+     */
+    private static String deployApp(String name, String templatePath) throws IOException {
+        System.out.println("Deploy " + name + " using openShift template: " + templatePath);
+
+        String appName = "";
+        List<? extends HasMetadata> entities = OPENSHIFT.deploy(name, new File(templatePath));
+
+        Optional<String> first = entities.stream().filter(hm -> hm instanceof DeploymentConfig).map(hm -> (DeploymentConfig) hm)
+                .map(dc -> dc.getMetadata().getName()).findFirst();
+        if (first.isPresent()) {
+            appName = first.get();
+        }
```

Otherwise? The `OpenShiftTestAssistant.deployApplication` method has an external source of app name, but here, we should probably throw an exception.

---

Ok.